mysql-test/suite/innodb/t/buffer_pool.test

```diff
@@ -0,0 +1,20 @@
+--source include/have_innodb.inc
+
+SET GLOBAL innodb_max_purge_lag_wait=0;
+CREATE TEMPORARY TABLE t1(a SMALLINT PRIMARY KEY) ENGINE=InnoDB;
+BEGIN;
+SET @read0=
+(SELECT CAST(variable_value AS INTEGER) FROM information_schema.global_status
+WHERE variable_name = 'innodb_buffer_pool_read_requests');
+INSERT INTO t1 VALUES (0),(1),(2),(3),(4),(5),(6),(7),(8),(9);
+SET @read1=
+(SELECT CAST(variable_value AS INTEGER) FROM information_schema.global_status
+WHERE variable_name = 'innodb_buffer_pool_read_requests');
+INSERT INTO t1 VALUES (10),(11),(12),(13),(14),(15),(16),(17),(18),(19);
+SET @read2=
+(SELECT CAST(variable_value AS INTEGER) FROM information_schema.global_status
+WHERE variable_name = 'innodb_buffer_pool_read_requests');
+COMMIT;
+
+DROP TABLE t1;
+SELECT @read1-@read0, @read2-@read1;
```

Comment:
No concurrent background operation should be possible. Specifically, the statement
```sql
SET GLOBAL innodb_max_purge_lag_wait=0;
```
will wait for any history to be purged. There should be no updates of InnoDB persistent statistics either. Change buffer merge in the background was removed in b42294bc6409794bdbd2051b32fa079d81cea61d, and the change buffer was disabled by default in 5c46751f238ee8dcef1e718ac5f63952bff5d09d.

Because the test is creating a `TEMPORARY TABLE`, there will be no background activity related to that.

You can see in https://buildbot.mariadb.org/#/grid?branch=refs%2Fpull%2F3479%2Fmerge that this test has never failed for the 3 pushes so far, on any platform.

Reply:
Thanks for the explanation. Let's hope that we could maintain the result file stability.

On a side note, it looks like purge is entirely disabled for temp table in MariaDB. It could be interesting to discuss this topic sometime but is unrelated to this patch.